src/interface/mod.rs

```diff
@@ -0,0 +1,119 @@
+use alloy::{contract::Error, providers::RootProvider, sol, transports::http::Http};
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+
+use self::IAggregatorV3Interface::IAggregatorV3InterfaceInstance;
+
+sol!(
+    #[allow(missing_docs)]
+    #[sol(rpc)]
+    IAggregatorV3Interface,
+    "src/abi/IAggregatorV3Interface.json"
+);
+
+pub trait Serializable {
+    fn to_bin(&self) -> Result<Vec<u8>, Box<bincode::ErrorKind>>;
+    fn from_bin(data: Vec<u8>) -> Result<Self, Box<bincode::ErrorKind>>
+    where
+        Self: Sized;
+}
+
+#[derive(Clone)]
+pub struct ChainlinkContract {
+    pub contract: IAggregatorV3InterfaceInstance<Http<Client>, RootProvider<Http<Client>>>,
+    pub decimals: u8,
+}
+
+/// The latest price received for this symbol.
+/// This data is directly retrieved from the underlying contract.
+#[derive(Deserialize, Serialize)]
+pub struct PriceData {
+    /// Id of the submission by the aggregator
+    pub round_id: u128,
+    /// Answered in round
+    pub answered_in_round: u128,
+    /// Timestamp for when the aggregator started collecting data
+    pub started_at: String,
+    /// Timestamp for when the aggregator posted the price update
+    pub updated_at: String,
+    /// Price of the asset         
+    pub price: f64,
+}
+
+impl Serializable for PriceData {
```

Comment:
`Serializable` trait and impl look useless here as you can invoke `bincode::serialize()` directly on `PriceData`.